packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapHeatmapController.m

```diff
@@ -0,0 +1,186 @@
+
+// The map view owned by GoogmeMapController.
+@property(nonatomic, weak) GMSMapView *mapView;
+
+@end
+
+// Static wrapper for interpreting heatmap options.
+static void InterpretHeatmapOptions(FLTGoogleMapHeatmapController *self,
+                                    NSDictionary<NSString *, id> *options) {
+  [self interpretHeatmapOptions:options];
+}
+
+@implementation FLTGoogleMapHeatmapController
+- (instancetype)initWithHeatmapTileLayer:(GMUHeatmapTileLayer *)heatmapTileLayer
+                                 mapView:(GMSMapView *)mapView
+                                 options:(NSDictionary<NSString *, id> *)options {
+  self = [super init];
+  if (self) {
+    _heatmapTileLayer = heatmapTileLayer;
+    _mapView = mapView;
+    InterpretHeatmapOptions(self, options);
+  }
+  return self;
+}
+
+- (void)removeHeatmap {
+  _heatmapTileLayer.map = nil;
+}
+
+- (void)clearTileCache {
+  [_heatmapTileLayer clearTileCache];
+}
+
+- (void)setWeightedData:(NSArray<GMUWeightedLatLng *> *)weightedData {
+  _heatmapTileLayer.weightedData = weightedData;
+}
+
+- (void)setGradient:(GMUGradient *)gradient {
+  _heatmapTileLayer.gradient = gradient;
+}
+
+- (void)setOpacity:(double)opacity {
+  _heatmapTileLayer.opacity = opacity;
+}
+
+- (void)setRadius:(int)radius {
+  _heatmapTileLayer.radius = radius;
+}
+
+- (void)setMinimumZoomIntensity:(int)intensity {
+  _heatmapTileLayer.minimumZoomIntensity = intensity;
+}
+
+- (void)setMaximumZoomIntensity:(int)intensity {
+  _heatmapTileLayer.maximumZoomIntensity = intensity;
+}
+
+- (void)setMap {
+  _heatmapTileLayer.map = _mapView;
+}
+
+- (void)interpretHeatmapOptions:(NSDictionary<NSString *, id> *)data {
+  id weightedData = data[kHeatmapDataKey];
+  if ([weightedData isKindOfClass:[NSArray class]]) {
+    [self setWeightedData:[FLTGoogleMapJSONConversions weightedDataFromArray:weightedData]];
+  }
+
+  id gradient = data[kHeatmapGradientKey];
+  if ([gradient isKindOfClass:[NSDictionary class]]) {
+    [self setGradient:[FLTGoogleMapJSONConversions gradientFromDictionary:gradient]];
+  }
+
+  id opacity = data[kHeatmapOpacityKey];
+  if ([opacity isKindOfClass:[NSNumber class]]) {
+    [self setOpacity:[opacity doubleValue]];
+  }
+
+  id radius = data[kHeatmapRadiusKey];
+  if ([radius isKindOfClass:[NSNumber class]]) {
+    [self setRadius:[radius intValue]];
+  }
+
+  id minimumZoomIntensity = data[kHeatmapMinimumZoomIntensityKey];
+  if ([minimumZoomIntensity isKindOfClass:[NSNumber class]]) {
+    [self setMinimumZoomIntensity:[minimumZoomIntensity intValue]];
+  }
+
+  id maximumZoomIntensity = data[kHeatmapMaximumZoomIntensityKey];
+  if ([maximumZoomIntensity isKindOfClass:[NSNumber class]]) {
+    [self setMaximumZoomIntensity:[maximumZoomIntensity intValue]];
+  }
+
+  // The map must be set each time for options to update.
+  [self setMap];
+}
+- (NSDictionary<NSString *, id> *)getHeatmapInfo {
+  NSMutableDictionary *options = [@{} mutableCopy];
```

Comment:
There's no conditional logic here, so this doesn't need to be a mutable dictionary. It can just:
```objc
return @{
  kHeatmapDataKey: [FLTGoogleMapJSONConversions arrayFromWeightedData:_heatmapTileLayer.weightedData],
  kHeatmapGradientKey: ...,
  ...
};
```